### DISCUSSION TITLE
EILI5?  Examples for dummies?

### DISCUSSION THREAD

**Discussion by evanrich** [docker/metadata-action#393] [Q&A]
I'm probably dense, but is there like an ELI5 guide or some examples somewhere?  I'm not grasping the (well written) docs.  

I've got the following actions:
```
   - name: Build Docker image
      run: docker build -t harbor.server.tld/library/package:latest ./Docker/

    - name: Minor version for each merge
      id: taggerFinal
      uses: anothrNick/github-tag-action@1.67.0
      env:
        GITHUB_TOKEN: ${{ secrets.REPO_PUSH_TOKEN }}
        WITH_V: true
        GIT_API_TAGGING: false

    - name: tag with name
      run: docker tag  harbor.server.tld/library/package:latest harbor.server.tld/library/package:${{ steps.taggerFinal.outputs.new_tag }}
```

I've replaced it with the following from the [docker suggested code](https://docs.docker.com/build/ci/github-actions/manage-tags-labels/):

```
jobs:
  docker:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v4
      - name: Docker meta
        id: meta
        uses: docker/metadata-action@v5
        with:
          # list of Docker images to use as base name for tags
          images: |
            name/app
            ghcr.io/username/app            
          # generate Docker tags based on the following events/attributes
          tags: |
            type=schedule
            type=ref,event=branch
            type=ref,event=pr
            type=semver,pattern={{version}}
            type=semver,pattern={{major}}.{{minor}}
            type=semver,pattern={{major}}
            type=sha            
      - name: Login to Docker Hub
        if: github.event_name != 'pull_request'
        uses: docker/login-action@v3
        with:
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}
      - name: Build and push
        uses: docker/build-push-action@v5
        with:
          context: .
          push: ${{ github.event_name != 'pull_request' }}
          tags: ${{ steps.meta.outputs.tags }}
          labels: ${{ steps.meta.outputs.labels }}
```
but i have questions..  How does the semver know to increment?  how does latest get applied? when I tried running this, I only got a sha and branch name:
![image](https://github.com/docker/metadata-action/assets/518320/e47eba9c-7509-4b98-bdd0-fe97b34f35ef)
Do I still need to run semver to create a tag first?  thanks!

**Comment by evanrich** [comment]
fwiw, i tried running this after creating a new tag, v1.13.0, but all that got generated was 
![image](https://github.com/docker/metadata-action/assets/518320/52845eee-78c7-407c-98e7-0da6fbef17fa)

does it not like the "v" in front of the semver?

**Comment by peterweissdk** [comment]
I think you need to add the flavor option.

- name: Docker meta
With:
  flavor: |
  latest=false
